src/main/java/seedu/address/model/techstack/TechStack.java

```diff
@@ -25,6 +26,11 @@ public TechStack(String techStackName) {
         this.techStackName = techStackName;
     }
 
+    public void setRating(int rating) {
```

Comment:
This unfortunately violates the constraint that `TechStack` should be immutable. I think it would be better if you made a copy of `TechStack` to edit it, which you already are doing for the rest of `Contact`'s fields anyway.